File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.cpp
class Solution {
public:
    int firstOcc(vector<int>arr,int target){
        int low = 0;
        int high = arr.size()-1;
        int firstOccurence = -1;

        while(low <= high){
            int mid = low + (high-low)/2;
            
            if(arr[mid] == target){
                firstOccurence = mid;
                high = mid-1;
            }else if(arr[mid] < target){
                low = mid+1;
            }else{
                high = mid-1;
            }
        }
        return firstOccurence;
    }
    int lastOcc(vector<int>arr,int target){
        int low = 0;
        int high = arr.size()-1;
        int lastOccurence = -1;

        while(low <= high){
            int mid = low + (high-low)/2;
            
            if(arr[mid] == target){
                lastOccurence = mid;
                low = mid+1;
            }else if(arr[mid] < target){
                low = mid+1;
            }else{
                high = mid-1;
            }
        }
        return lastOccurence;
    }
    vector<int> searchRange(vector<int>& nums, int target) {

        int first = firstOcc(nums,target);
        int last = lastOcc(nums,target);

        vector<int> ans = {first,last};

        return ans;
    }
};

File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.java
class Solution {
    int firstOcc(int [] nums,int target){
        int n = nums.length;
        int low = 0;
        int high = n-1;
        int idx = -1;

        while(low <= high){
            int mid = low + (high-low)/2;

            if(nums[mid] == target){
                idx = mid;
                high = mid-1;
            }else if(nums[mid] < target)low = mid+1;
            else high = mid-1;
        }

        return idx;
    }
    int lastOcc(int [] nums, int target){
        int n = nums.length;
        int low = 0;
        int high = n-1;
        int idx = -1;

        while(low <= high){
            int mid = low + (high-low)/2;

            if(nums[mid] == target){
                idx = mid;
                low = mid+1;
            }else if(nums[mid] < target)low = mid+1;
            else high = mid-1;
        }

        return idx;
    }
    public int[] searchRange(int[] nums, int target) {
        int []res = new int[2];
        
        res[0] = firstOcc(nums,target);
        res[1] = lastOcc(nums,target);

        return res;
    }
}